src/main/java/edu/kis/vh/nursery/DefaultCountingOutRhymer.java

```diff
@@ -39,4 +39,6 @@ protected int countOut() {
     protected int getTotal() {
         return stack.getTotal();
     }
-}
\ No newline at end of file
+}
+
+// answer 3.1.18 - all three classes break the rule, because they have no common interface.
```

Comment:
Only one rhymer breaks insulation between abstraction and implementation hierarchies.